auxx/src/Lang/DisplayError.hs

```diff
@@ -0,0 +1,128 @@
+                                               punctuate, squotes, text, vcat, yellow, (<$>), (<+>))
+
+import           Lang.Argument (ArgumentError (..), ProcError (..), TypeError (..), TypeName (..),
+                                isEmptyArgumentError)
+import           Lang.Interpreter (EvalError (..))
+import           Lang.Lexer (Token (..))
+import           Lang.Name (Name)
+import           Lang.Parser (ParseError (..))
+
+
+highlight :: Doc -> Doc
+highlight = bold . yellow
+
+nameToDoc :: Name -> Doc
+nameToDoc = text . T.unpack . toLazyText . build
+
+ppTypeName :: TypeName -> Doc
+ppTypeName (TypeName name)         = text (unpack name)
+ppTypeName (TypeNameEither tn tn') = ppTypeName tn <+> char '|' <+> ppTypeName tn'
+
+ppTypeError :: TypeError -> Doc
+ppTypeError TypeError{..} =
+        "Couldn't match expected type" <+> (bold . yellow $ ppTypeName teExpectedType)
+    <+> "with actual value"            <+> (bold . yellow $ show teActualValue)
+    <>  "!"
+
+ppArgumentError :: ArgumentError -> Doc
+ppArgumentError ae@ArgumentError{..} =
+    if isEmptyArgumentError ae
+    then empty
+    else vcat $ missingKeysDoc <> irrelevantKeysDoc <> irrelevantPosDoc
+  where
+    setToDoc = hcat . punctuate comma . map (highlight . nameToDoc) . S.toList
+    missingKeysDoc =
+        if S.null aeMissingKeys
+        then []
+        else ["Missing keys:" <+> setToDoc aeMissingKeys]
+    irrelevantKeysDoc =
+        if S.null aeIrrelevantKeys
+        then []
+        else ["Irrelevant keys:" <+> setToDoc aeIrrelevantKeys]
+    irrelevantPosDoc =
+        if aeIrrelevantPos == 0
+        then []
+        else ["Irrelevant positional argument:" <+> (highlight . show) aeIrrelevantPos]
+
+ppProcError :: ProcError -> Doc
+ppProcError ProcError{..} = ppArgumentError peArgumentError <$> typeErrorsDoc
+  where
+    typeErrorsDoc =
+        if S.null peTypeErrors
+        then empty
+        else "Following type errors occured:" <$>
+             (indent 2 . hcat . map ppTypeError . S.toList) peTypeErrors
+
+ppEvalError :: EvalError -> Doc
+ppEvalError (CommandNotSupported name) =
+        "Command" <+> (squotes . highlight . nameToDoc) name
+    <+> "is unavailable. Try" <+> (squotes . highlight) "help"
+ppEvalError (InvalidArguments name procError) =
+        "Invalid arguments for" <+> (squotes . highlight . nameToDoc) name <> ":"
+    <$> indent 2 (ppProcError procError)
+
+
+renderLine :: Int -> Int -> Text -> Doc
+renderLine start end str = (text $ unpack str)
+  <$> highlight (indent start . hcat . replicate (end - start) $ char '^')
+
+renderFullLine :: Text -> Doc
+renderFullLine str = renderLine 0 (length str) str
+
+ppParseError :: ParseError -> Doc
+ppParseError (ParseError str (Report {..})) =
+      "Parse error at" <+> text (show span)
+  <$> "Unexpected" <+> text unconsumedDesc <> ", expected"
+  <+> hcat (punctuate (text ", or ") $ map (text . unpack) expected)
```

Comment:
This is the third time I see `text . unpack` here. Try doing something like this:

```
import qualified Text.PrettyPrint.ANSI.Leijen

text = Text.PrettyPrint.ANSI.Leijen.text . Universum.toText
```